File: psra/web/src/main/java/by/arhor/psra/web/controller/CourseController.scala
package by.arhor.psra.web.controller

import by.arhor.psra.dto.CourseDto
import by.arhor.psra.service.CourseService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE
import org.springframework.web.bind.annotation.{CrossOrigin, GetMapping, PathVariable, RequestMapping, RestController}

@CrossOrigin
@RestController
@RequestMapping(path = Array("/courses"))
class CourseController(@Autowired service: CourseService) {

  @GetMapping(path = Array("/{id}"), produces = Array(APPLICATION_JSON_UTF8_VALUE))
  def getGalleryById(@PathVariable("id") id: String): CourseDto = service.findOne(id)

}

File: psra/web/src/main/java/by/arhor/psra/web/controller/CourseController.java
package by.arhor.psra.web.controller;

import by.arhor.psra.dto.CourseDto;
import by.arhor.psra.service.CourseService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE;

@CrossOrigin
@RestController
@RequestMapping(path = "/courses")
public class CourseController {

  private final CourseService service;

  @Autowired
  public CourseController(CourseService service) {
    this.service = service;
  }

  @GetMapping(path = "/{id}", produces = APPLICATION_JSON_UTF8_VALUE)
  public CourseDto getGalleryById(@PathVariable("id") String id) {
    return service.findOne(id);
  }
}
